// angular-form-builder/Gruntfile.js
module.exports = function(grunt) {
  grunt.config.init({
    compass: {
      example: {
        options: {
          sassDir: 'example',
          cssDir: 'example',
          outputStyle: 'compressed'
        }
      },
      src: {
        options: {
          sassDir: 'src',
          cssDir: 'dist',
          outputStyle: 'compressed'
        }
      }
    },
   concat: {
    builder: {
      src: ['src/*.js'],
      dest: 'dist/angular-form-builder.js',
    },
    components: {
      src: ['components/*.js'],
      dest: 'dist/angular-form-builder-components.js',
    }
  },
    uglify: {
      build: {
        files: {
          'dist/angular-form-builder.min.js': 'dist/angular-form-builder.js',
          'dist/angular-form-builder-components.min.js': 'dist/angular-form-builder-components.js'
        }
      }
    },
    watch: {
      compass: {
        files: ['example/*.scss', 'src/*.scss'],
        tasks: ['compass'],
        options: {
          spawn: false
        }
      },
      js: {
        files: ['src/*.js', 'components/*.js', 'example/*.js'],
        tasks: ['js'],
        options: {
          spawn: false
        }
      }
    },
    connect: {
      server: {
        options: {
          protocol: 'http',
          hostname: '*',
          port: 8000,
          base: '.'
        }
      }
    },
    karma: {
      min: {
        configFile: 'test/karma-min.config.js'
      },
      source: {
        configFile: 'test/karma.config.js'
      }
    }
  });
  grunt.registerTask('dev', ['compass', 'concat', 'connect', 'watch']);
  grunt.registerTask('build', ['compass', 'concat', 'uglify']);
  grunt.registerTask('test', ['karma']);
  grunt.loadNpmTasks('grunt-contrib-compass');
  grunt.loadNpmTasks('grunt-contrib-concat');
  grunt.loadNpmTasks('grunt-contrib-watch');
  grunt.loadNpmTasks('grunt-contrib-connect');
  grunt.loadNpmTasks('grunt-karma');
  return grunt.loadNpmTasks('grunt-contrib-uglify');
};
